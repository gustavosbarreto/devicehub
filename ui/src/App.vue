<template>
<v-app>
    <v-navigation-drawer v-if="isLoggedIn" :clipped="false" fixed v-model="drawer" :mini-variant="false" enable-resize-watcher app>
        <v-container>
            <div class="text-center">
                <v-icon>mdi-console</v-icon>
                <h2 style="font-family: monospace">ShellHub</h2>
                <span class="overline">beta</span>
            </div>
        </v-container>
        <v-list>
            <v-list-item v-for="item in items" :key="item.title" :to="item.path" two-line>
                <v-list-item-action>
                    <v-icon v-text="item.icon"></v-icon>
                </v-list-item-action>

                <v-list-item-content>
                    <v-list-item-title v-text="item.title"></v-list-item-title>
                </v-list-item-content>
            </v-list-item>
        </v-list>
    </v-navigation-drawer>
    <v-app-bar app elevate-on-scroll color="primary" class="pl-3 pr-4" v-if="isLoggedIn">
        <v-menu transition="scale-transition" origin="top left">
            <template v-slot:activator="{ on }">
                <v-chip v-on="on">
                    <v-icon left>mdi-server</v-icon>
                    My Device Fleet
                    <v-icon right>mdi-chevron-down</v-icon>
                </v-chip>
            </template>
        </v-menu>
        <v-spacer></v-spacer>
        <v-chip>
            <v-icon>help</v-icon>
        </v-chip>
        <v-chip>
            <v-icon>notifications</v-icon>
        </v-chip>
        <v-menu transition="scale-transition" origin="top right">
            <template v-slot:activator="{ on }">
                <v-chip v-on="on">
                    <v-icon left>mdi-account</v-icon>
                    {{ $store.getters["auth/currentUser"] }}
                    <v-icon right>mdi-chevron-down</v-icon>
                </v-chip>
            </template>

            <v-card>
                <v-list-item three-line>
                    <v-list-item-content>
                        <v-list-item-title class="mb-1">Tenant ID</v-list-item-title>
                        <v-list-item-subtitle>
                            <v-chip>
                                <span>
                                    {{ tenant }}
                                </span>
                                <v-icon right v-clipboard="tenant" v-clipboard:success="() => { copySnack = true; }">mdi-content-copy</v-icon>
                            </v-chip>
                        </v-list-item-subtitle>
                    </v-list-item-content>
                </v-list-item>

                <v-card-actions>
                    <v-btn small text @click="logout()">
                        Logout
                    </v-btn>
                </v-card-actions>
            </v-card>
        </v-menu>
    </v-app-bar>
    <v-content>
        <v-container class="pa-8" fluid>
            <router-view></router-view>
        </v-container>
        <v-snackbar v-model="copySnack" :timeout=3000>Tenant ID copied to clipboard</v-snackbar>
    </v-content>
</v-app>
</template>

<script>
export default {
  name: "App",

  methods: {
    logout() {
      this.$store.dispatch("auth/logout").then(() => {
        this.$router.push("/login");
      });
    }
  },

  computed: {
    tenant() {
      return this.$store.getters["auth/tenant"];
    },

    isLoggedIn() {
      return this.$store.getters["auth/isLoggedIn"];
    }
  },

  data() {
    return {
      drawer: true,
      clipped: false,
      copySnack: false,
      items: [
        {
          icon: "dashboard",
          title: "Dashboard",
          path: "/"
        },
        {
          icon: "devices",
          title: "Devices",
          path: "/devices"
        },
        {
          icon: "history",
          title: "Sessions",
          path: "/sessions"
        }
      ]
    };
  }
};
</script>

<style>
.v-list-active {
  border-left: 4px solid var(--v-primary-base);
}

.text-shadow {
  text-shadow: #000 0 0 6px;
  color: transparent;
}
</style>
