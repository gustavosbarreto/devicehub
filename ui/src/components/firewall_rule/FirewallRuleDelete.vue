<template>
  <fragment>
    <v-tooltip bottom>
      <template #activator="{ on }">
        <span v-on="on">
          <v-icon
            :disabled="!isOwner"
            v-on="on"
            @click="dialog = !dialog"
          >
            delete
          </v-icon>
        </span>
      </template>

      <div>
        <span v-if="isOwner">
          Remove
        </span>

        <span v-else>
          You are not the owner of this namespace
        </span>
      </div>
    </v-tooltip>

    <v-dialog
      v-model="dialog"
      max-width="400"
    >
      <v-card>
        <v-card-title class="headline grey lighten-2 text-center">
          Are you sure?
        </v-card-title>

        <v-card-text class="mt-4 mb-3 pb-1">
          You are about to remove this firewall rule
        </v-card-text>

        <v-card-actions>
          <v-spacer />

          <v-btn
            text
            @click="dialog=!dialog"
          >
            Close
          </v-btn>

          <v-btn
            color="red darken-1"
            text
            @click="remove();"
          >
            Remove
          </v-btn>
        </v-card-actions>
      </v-card>
    </v-dialog>
  </fragment>
</template>

<script>

export default {
  name: 'FirewallRuleDelete',

  props: {
    id: {
      type: String,
      required: true,
    },
  },

  data() {
    return {
      dialog: false,
    };
  },

  computed: {
    isOwner() {
      return this.$store.getters['namespaces/owner'];
    },
  },

  methods: {
    async remove() {
      try {
        await this.$store.dispatch('firewallrules/remove', this.id);
        this.$store.dispatch('snackbar/showSnackbarSuccessAction', this.$success.firewallRuleDeleting);
        this.update();
      } catch {
        this.$store.dispatch('snackbar/showSnackbarErrorAction', this.$errors.snackbar.firewallRuleDeleting);
      }
    },

    update() {
      this.$emit('update');
      this.close();
    },

    close() {
      this.dialog = !this.dialog;
    },
  },
};
</script>
