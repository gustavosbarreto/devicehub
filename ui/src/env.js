export default {
  /* eslint-disable no-param-reassign */
  install(Vue) {
    const env = {
      isHosted: window.env.HOSTED === 'true',
    };

    Vue.env = env;
    Vue.prototype.$env = env;
  },
  /* eslint-enable no-param-reassign */
};
