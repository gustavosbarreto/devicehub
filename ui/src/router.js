import Vue from "vue";
import Router from "vue-router";
import Dashboard from "./views/Dashboard.vue";
import store from './store'

Vue.use(Router);

const router = new Router({
    mode: "history",
    base: process.env.BASE_URL,
    routes: [
        {
            path: "/",
            name: "dashboard",
            component: Dashboard
        },
        {
            path: "/devices",
            name: "devices",

            component: () =>
                import(/* webpackChunkName: "devices" */ "./views/Devices.vue")
        },
        {
            path: "/devices/:id",
            name: "featureDevice",
            component: () =>
                import("./views/FeatureDevice.vue") 
        },
        {
            path: "/sessions",
            name: "sessions",
            component: () =>
                import("./views/Sessions.vue")
        },
        {
            path: "/login",
            name: "login",
            component: () =>
                import("./views/Login.vue")
        }
    ]
});

router.beforeEach((to, from, next) => {
    if (to.path !== '/login') {
        if (store.getters['auth/isLoggedIn']) {
          return next()
        }

        return next(`/login?redirect=${to.path}`)
      } else {
        return next()
      }
})

export default router;