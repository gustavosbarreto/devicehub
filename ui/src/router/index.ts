import { RouteRecordRaw, createRouter, createWebHistory } from "vue-router";
import { envVariables } from "../envVariables";
import { store } from "@/store";

const Dashboard = () => import("@/views/Dashboard.vue");
const Devices = () => import("@/views/Devices.vue");
const DeviceList = () => import("@/components/Devices/DeviceList.vue");
const DevicePendingList = () => import("@/components/Devices/DevicePendingList.vue");
const DeviceRejectedList = () => import("@/components/Devices/DeviceRejectedList.vue");
const Containers = () => import("@/views/Containers.vue");
const ContainerList = () => import("@/components/Containers/ContainerList.vue");
const ContainerPendingList = () => import("@/components/Containers/ContainerPendingList.vue");
const ContainerRejectedList = () => import("@/components/Containers/ContainerRejectedList.vue");
const Connectors = () => import("@/views/Connectors.vue");
const detailsConnectors = () => import("@/views/ConnectorDetails.vue");
const DetailsDevice = () => import("@/views/DetailsDevice.vue");
const Sessions = () => import("@/views/Sessions.vue");
const DetailsSessions = () => import("@/views/DetailsSessions.vue");
const FirewallRules = () => import("@/views/FirewallRules.vue");
const PublicKeys = () => import("@/views/PublicKeys.vue");
const Settings = () => import("@/views/Settings.vue");
const SettingProfile = () => import("@/components/Setting/SettingProfile.vue");
const SettingNamespace = () => import("@/components/Setting/SettingNamespace.vue");
const SettingPrivateKeys = () => import("@/components/Setting/SettingPrivateKeys.vue");
const SettingTags = () => import("@/components/Setting/SettingTags.vue");
const SettingBilling = () => import("@/components/Setting/SettingBilling.vue");

export const routes: Array<RouteRecordRaw> = [
  {
    path: "/login",
    name: "login",
    meta: {
      layout: "LoginLayout",
      requiresAuth: false,
    },
    component: () => import("../views/Login.vue"),
  },
  {
    path: "/mfa-login",
    name: "MfaLogin",
    beforeEnter: (to, from, next) => {
      // Check if the user is coming from the login route
      if (from.name === "login") {
        // Allow access to MFA login if the user is coming from the login route
        next();
      } else {
        // Redirect to login if the user is not coming from the login route
        next({ name: "login" });
      }
    },
    meta: {
      layout: "LoginLayout",
      requiresAuth: false,
    },
    component: () => import("../views/MfaLogin.vue"),
  },
  {
    path: "/recover-mfa",
    name: "RecoverMfa",
    beforeEnter: (to, from, next) => {
      if (from.name === "MfaLogin") {
        next();
      } else {
        next({ name: "login" });
      }
    },
    meta: {
      layout: "LoginLayout",
      requiresAuth: false,
    },
    component: () => import("../components/AuthMFA/MfaRecover.vue"),
  },
  {
    path: "/recover-mfa/mail-sucessful",
    name: "RecoverMfaMsg",
    beforeEnter: (to, from, next) => {
      if (from.name === "RecoverMfa") {
        next();
      } else {
        next({ name: "login" });
      }
    },
    meta: {
      layout: "LoginLayout",
      requiresAuth: false,
    },
    component: () => import("../components/AuthMFA/MfaMailRecover.vue"),
  },
  {
    path: "/forgot-pass",
    name: "ForgotPassword",
    meta: {
      layout: "LoginLayout",
      requiresAuth: false,
    },
    component: () => import("../views/ForgotPassword.vue"),
  },
  {
    path: "/validation-account",
    name: "ValidationAccount",
    meta: {
      layout: "LoginLayout",
      requiresAuth: false,
    },
    component: () => import("../views/ValidationAccount.vue"),
  },
  {
    path: "/reset-mfa",
    name: "MfaResetValidation",
    meta: {
      layout: "LoginLayout",
      requiresAuth: false,
    },
    component: () => import("../views/MfaResetValidation.vue"),
  },
  {
    path: "/update-password",
    name: "UpdatePassword",
    meta: {
      layout: "LoginLayout",
      requiresAuth: false,
    },
    component: () => import("../views/UpdatePassword.vue"),
  },
  {
    path: "/sign-up",
    name: "SignUp",
    meta: {
      layout: "LoginLayout",
      requiresAuth: false,
    },
    component: () => import("../views/SignUp.vue"),
  },
  {
    path: "/confirm-account",
    name: "ConfirmAccount",
    meta: {
      layout: "LoginLayout",
      requiresAuth: false,
    },
    component: () => import("../views/ConfirmAccount.vue"),
  },
  {
    path: "/",
    name: "Dashboard",
    component: Dashboard,
  },
  {
    path: "/devices",
    name: "devices",
    component: Devices,
    redirect: {
      name: "listDevices",
    },
    children: [
      {
        path: "",
        name: "listDevices",
        component: DeviceList,
      },
      {
        path: "pending",
        name: "pendingDevices",
        component: DevicePendingList,
      },
      {
        path: "rejected",
        name: "rejectedDevices",
        component: DeviceRejectedList,
      },
    ],
  },
  {
    path: "/containers",
    name: "containers",
    beforeEnter: (to, from, next) => {
      if (!envVariables.isEnterprise && !envVariables.isCloud && envVariables.premiumPaywall) {
        store.commit("users/setShowPaywall", true);
      }
      next();
    },
    component: Containers,
    redirect: {
      name: "listContainers",
    },
    children: [
      {
        path: "",
        name: "listContainers",
        component: ContainerList,
      },
      {
        path: "pending",
        name: "pendingContainers",
        component: ContainerPendingList,
      },
      {
        path: "rejected",
        name: "rejectedContainers",
        component: ContainerRejectedList,
      },
    ],
  },
  {
    path: "/containers/connectors",
    name: "connectors",
    component: Connectors,
    beforeEnter: (to, from, next) => {
      if (!envVariables.isEnterprise && !envVariables.isCloud && envVariables.premiumPaywall) {
        store.commit("users/setShowPaywall", true);
      }
      next();
    },
  },
  {
    path: "/containers/connectors/:id",
    name: "detailsConnectors",
    component: detailsConnectors,
  },
  {
    path: "/device/:id",
    name: "detailsDevice",
    component: DetailsDevice,
  },
  {
    path: "/sessions",
    name: "Sessions",
    component: Sessions,
  },
  {
    path: "/sessions/:id",
    name: "detailsSession",
    component: DetailsSessions,
  },
  {
    path: "/firewall/rules",
    name: "firewalls",
    component: FirewallRules,
    beforeEnter: (to, from, next) => {
      if (!envVariables.isEnterprise && !envVariables.isCloud && envVariables.premiumPaywall) {
        store.commit("users/setShowPaywall", true);
      }
      next();
    },
  },
  {
    path: "/sshkeys/public-keys",
    name: "publicKeys",
    component: PublicKeys,
  },
  {
    path: "/settings",
    name: "settings",
    component: Settings,
    redirect: {
      name: "profileSettings",
    },
    children: [
      {
        path: "profile",
        name: "profileSettings",
        component: SettingProfile,
      },
      {
        path: "namespace-manager",
        name: "namespaceSettings",
        component: SettingNamespace,
      },
      {
        path: "private-keys",
        name: "privateKeysSettings",
        component: SettingPrivateKeys,
      },
      {
        path: "tags",
        name: "tagsSettings",
        component: SettingTags,
      },
      {
        path: "billing",
        name: "billingSettings",
        beforeEnter: (to, from, next) => {
          const enabled = envVariables.billingEnable;
          if (enabled) {
            next();
          } else {
            next("/404");
          }
        },
        component: SettingBilling,
      },
    ],
  },
  {
    path: "/:catchAll(.*)",
    redirect: { name: "NotFound" },
  },
  {
    path: "/404",
    name: "NotFound",
    component: () => import("../views/NotFound.vue"),
  },
];

export const router = createRouter({
  history: createWebHistory(),
  routes,
});

router.beforeEach(async (to, from, next) => {
  const isLoggedIn = store.getters["auth/isLoggedIn"];
  const layout = to.meta.layout || "AppLayout";
  const requiresAuth = to.meta.requiresAuth ?? true;

  await store.dispatch("layout/setLayout", layout);

  // Redirect to the appropriate page based on authentication status
  if (requiresAuth && !isLoggedIn) {
    next({ name: "login" }); // Redirect to login page if authentication is required and user is not logged in
  } else if (to.name === "login" && isLoggedIn) {
    next({ path: "/" }); // Redirect from login page to home if user is already logged in
  } else {
    next(); // Continue with the original navigation
  }
});
