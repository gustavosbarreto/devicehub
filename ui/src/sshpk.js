const NodeRSA = require("node-rsa");
const sshpk = require("sshpk");

const createSignatureOfPrivateKey = (
  privateKeyData,
  username,
) => {
  const key = NodeRSA(privateKeyData);
  key.setOptions({ signingScheme: "pkcs1-sha1" });
  const signature = encodeURIComponent(key.sign(username, "base64"));
  return signature;
};

const createKeyFingerprint = (privateKeyData) => {
  const key = sshpk.parsePrivateKey(privateKeyData);
  const fingerprint = key.fingerprint("md5").toString("hex");
  return fingerprint;
};

const parsePrivateKey = (privateKey) => {
  const key = sshpk.parsePrivateKey(privateKey);
  return key;
};

const parseKey = (key) => {
  const parsedKey = sshpk.parseKey(key);
  return parsedKey;
};

const convertKeyToFingerprint = (privateKey) => {
  const fingerprint = sshpk.parsePrivateKey(privateKey).fingerprint("md5");
  return fingerprint;
};

const createSignerAndUpdate = (privateKey, username) => {
  const signer = privateKey.createSign("sha512");
  signer.update(username);
  return signer.sign().toString();
};

window.global.createSignatureOfPrivateKey = createSignatureOfPrivateKey;
window.global.createKeyFingerprint = createKeyFingerprint;
window.global.convertKeyToFingerprint = convertKeyToFingerprint;
window.global.createSignerAndUpdate = createSignerAndUpdate;
window.global.parsePrivateKey = parsePrivateKey;
window.global.parseKey = parseKey;
