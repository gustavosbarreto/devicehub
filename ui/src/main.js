import Vue from 'vue';

import Snackbar from '@/components/snackbar/Snackbar';

import App from './App';
import router from './router';
import store from './store';
import env from './env';
import './vee-validate';
import vuetify from './plugins/vuetify';

Vue.config.productionTip = false;

Vue.component('Snackbar', Snackbar);

Vue.use(require('vue-moment'));

Vue.use(env);

new Vue({
  vuetify,
  router,
  store,
  render: (h) => h(App),
}).$mount('#app');
