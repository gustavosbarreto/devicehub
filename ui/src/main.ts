import { createApp } from "vue";
import * as Sentry from "@sentry/vue";
import { BrowserTracing } from "@sentry/tracing";
import { envVariables } from "./envVariables";
import vuetify from "./plugins/vuetify";
import { key, store } from "./store";
import { router } from "./router";
import App from "./App.vue";

import { loadFonts } from "./plugins/webfontloader";

import SnackbarComponent from "./components/Snackbar/Snackbar.vue";
import { SnackbarPlugin } from "./plugins/snackbar";

const app = createApp(App);

Sentry.init({
  app,
  dsn: envVariables.sentryDsn || "",
  integrations: [
    new BrowserTracing({
      routingInstrumentation: Sentry.vueRouterInstrumentation(router),
    }),
  ],
  tracesSampleRate: 1.0,
  hooks: ["activate", "create", "destroy", "mount", "update"],
  timeout: 500,
  release: envVariables.version || "latest",
});
Sentry.setTag("project", "shellhub-ui");

loadFonts();

app.use(vuetify);
app.use(router);
app.use(store, key);
app.use(SnackbarPlugin);
app.component("SnackbarComponent", SnackbarComponent);
app.mount("#app");
