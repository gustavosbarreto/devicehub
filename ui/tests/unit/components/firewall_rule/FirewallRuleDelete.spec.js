import Vuex from 'vuex';
import { mount, createLocalVue } from '@vue/test-utils';
import Vuetify from 'vuetify';
import FirewallRuleDelete from '@/components/firewall_rule/FirewallRuleDelete';
import { actions, authorizer } from '../../../../src/authorizer';

describe('FirewallRuleDelete', () => {
  const localVue = createLocalVue();
  const vuetify = new Vuetify();
  localVue.use(Vuex);

  document.body.setAttribute('data-app', true);

  let wrapper;

  const role = ['owner', 'operator'];

  const hasAuthorization = {
    owner: true,
    operator: false,
  };

  const tests = [
    {
      description: 'Icon',
      props: {
        id: 'a582b47a42d',
        show: false,
      },
      data: {
        action: 'remove',
      },
      template: {
        'firewallRuleDelete-dialog': false,
        'close-btn': false,
        'remove-btn': false,
      },
    },
    {
      description: 'Dialog',
      props: {
        id: 'a582b47a42d',
        show: true,
      },
      data: {
        action: 'remove',
      },
      template: {
        'firewallRuleDelete-card': true,
        'close-btn': true,
        'remove-btn': true,
      },
    },
  ];

  const storeVuex = (currentrole) => new Vuex.Store({
    namespaced: true,
    state: {
      currentrole,
    },
    getters: {
      'auth/role': (state) => state.currentrole,
    },
    actions: {
      'firewallrules/remove': () => {},
      'snackbar/showSnackbarSuccessAction': () => {},
      'snackbar/showSnackbarErrorAction': () => {},
    },
  });

  tests.forEach((test) => {
    role.forEach((currentrole) => {
      describe(`${test.description} ${currentrole}`, () => {
        beforeEach(() => {
          wrapper = mount(FirewallRuleDelete, {
            store: storeVuex(currentrole),
            localVue,
            stubs: ['fragment'],
            propsData: { id: test.props.id, show: test.props.show },
            vuetify,
            mocks: {
              $authorizer: authorizer,
              $actions: actions,
            },
          });
        });

        ///////
        // Component Rendering
        //////

        it('Is a Vue instance', () => {
          expect(wrapper).toBeTruthy();
        });
        it('Renders the component', () => {
          expect(wrapper.html()).toMatchSnapshot();
        });

        ///////
        // Data checking
        //////

        it('Receive data in props', () => {
          Object.keys(test.props).forEach((item) => {
            expect(wrapper.vm[item]).toEqual(test.props[item]);
          });
        });
        it('Compare data with default value', () => {
          Object.keys(test.data).forEach((item) => {
            expect(wrapper.vm[item]).toEqual(test.data[item]);
          });
        });
        it('Process data in the computed', () => {
          expect(wrapper.vm.hasAuthorization).toEqual(hasAuthorization[currentrole]);
        });

        //////
        // HTML validation
        //////

        it('Renders the template with data', () => {
          Object.keys(test.template).forEach((item) => {
            if (!hasAuthorization[currentrole] && currentrole === 'operator' && test.props.show) {
              expect(wrapper.find(`[data-test="${item}"]`).exists()).toBe(!test.template[item]);
            } else {
              expect(wrapper.find(`[data-test="${item}"]`).exists()).toBe(test.template[item]);
            }
          });
        });
      });
    });
  });
});
