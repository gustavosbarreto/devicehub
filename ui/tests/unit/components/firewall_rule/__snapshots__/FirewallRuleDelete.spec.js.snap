// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirewallRuleDelete Dialog operator Renders the component 1`] = `
<fragment-stub><span><div data-test="delete-item" class="v-list-item__title">
          Remove
        </div></span><span class="v-tooltip v-tooltip--bottom"><!----></span><span><i aria-hidden="true" data-test="delete-icon" class="v-icon notranslate v-icon--disabled v-icon--left material-icons theme--light">delete</i></span>
  <div role="dialog" class="v-dialog__container">
    <!---->
  </div>
</fragment-stub>
`;

exports[`FirewallRuleDelete Dialog owner Renders the component 1`] = `
<fragment-stub><span><div data-test="delete-item" class="v-list-item__title">
          Remove
        </div></span><span class="v-tooltip v-tooltip--bottom"><!----></span><span><i aria-hidden="true" data-test="delete-icon" class="v-icon notranslate v-icon--left material-icons theme--light">delete</i></span>
  <div role="dialog" class="v-dialog__container"></div>
</fragment-stub>
`;

exports[`FirewallRuleDelete Icon operator Renders the component 1`] = `
<fragment-stub><span><div data-test="delete-item" class="v-list-item__title">
          Remove
        </div></span><span class="v-tooltip v-tooltip--bottom"><!----></span><span><i aria-hidden="true" data-test="delete-icon" class="v-icon notranslate v-icon--disabled v-icon--left material-icons theme--light">delete</i></span>
  <div role="dialog" class="v-dialog__container">
    <!---->
  </div>
</fragment-stub>
`;

exports[`FirewallRuleDelete Icon owner Renders the component 1`] = `
<fragment-stub><span><div data-test="delete-item" class="v-list-item__title">
          Remove
        </div></span><span class="v-tooltip v-tooltip--bottom"><!----></span><span><i aria-hidden="true" data-test="delete-icon" class="v-icon notranslate v-icon--left material-icons theme--light">delete</i></span>
  <div role="dialog" class="v-dialog__container">
    <!---->
  </div>
</fragment-stub>
`;
